### PR TITLE
docs/node: Refer to Oasis Core 23.0.x series everywhere

### PR DESCRIPTION
**Pull request by amela**
Preview: https://deploy-preview-603--trusting-archimedes-14c863.netlify.app/node/mainnet/eden-upgrade

### CODE REVIEW

File: docs/node/mainnet/eden-upgrade.md [hunk 1]
```diff
@@ -214,15 +214,15 @@ Ubuntu versions (or other distributions using older system libraries).
 [systemd]: https://systemd.io/
 [Supervisor]: http://supervisord.org/
 [Wipe state]: ../run-your-node/maintenance/wiping-node-state.md#state-wipe-and-keep-node-identity
-[23.0.3]: https://github.com/oasisprotocol/oasis-core/releases/tag/v23.0.3
+[23.0.x]: https://github.com/oasisprotocol/oasis-core/releases/tag/v23.0.x
```
Comment (kostko) [inline docs/node/mainnet/eden-upgrade.md:217]: This link needs to point to an actual release, can use the latest one, currently 23.0.4.

Comment (matevz) [reply docs/node/mainnet/eden-upgrade.md:217]: I agree, @amela. Let's avoid writing 23.0.x and just replace 23.0.3 with 23.0.4 (and 23.0.5, if we make another release before the upgrade) except in my next comment below.

File: docs/node/mainnet/eden-upgrade.md [hunk 2]
```diff
@@ -214,15 +214,15 @@ Ubuntu versions (or other distributions using older system libraries).
...
 [rosetta-gw-2.6.0]:
   https://github.com/oasisprotocol/oasis-rosetta-gateway/releases/tag/v2.6.0
 
 ### Configuration Changes
 
 :::info
 
-To see the full extent of the changes examine the [Change Log] of the 23.0.3
+To see the full extent of the changes examine the [Change Log] of the 23.0.x
```
Comment (matevz) [inline docs/node/mainnet/eden-upgrade.md:225]: Here, we probably want audience to read about changes that the new major version of Oasis Core brought and not the specific patch version, right?

```suggestion
To see the full extent of the changes examine the [Change Log] of the 23.0
```

**Review by matevz — CHANGES_REQUESTED** [review]

**Comment by matevz** [comment]
Superseeded by https://github.com/oasisprotocol/docs/pull/611.

**Pull request closed without merging**